saleor/graphql/order/dataloaders.py

```diff
@@ -35,6 +35,19 @@ def batch_load(self, keys):
         return [orders.get(order_id) for order_id in keys]
 
 
+class OrderByNumberLoader(DataLoader):
+    context_key = "order_by_number"
+
+    def batch_load(self, keys):
+        orders = Order.objects.using(self.database_connection_name).filter(
+            number__in=keys
+        )
+        orders_by_number = defaultdict(Order)
+        for order in orders:
+            orders_by_number[order.number] = order
```

Comment:
Instead of that, you can just do: `Order.objects.in_bulk(field_name="number")`

Reply:
Much cleaner, thanks!